hdfs/src/main/java/io/ddf/hdfs/HDFSDDF.java

```diff
@@ -53,20 +53,21 @@ private void initialize() throws DDFException {
         }
         // Check directory or file.
         HDFSDDFManager hdfsDDFManager = this.getManager();
+        mIsDir = hdfsDDFManager.isDir(this);
         // Check dataformat.
         if (options != null && options.containsKey("format")) {
             try {
                 String format = options.get("format").toUpperCase();
                 format = format.equals("PARQUET") ? "PQT" : format;
-                mDataFormat = DataFormat.valueOf(options.get("format"));
+                mDataFormat = DataFormat.valueOf(format);
             } catch (IllegalArgumentException e) {
-                mIsDir = hdfsDDFManager.isDir(this);
+                // TODO: Disable automatic format choosing, or put it under a convenience flag.
```

Comment:
It's moved above so that it always run. Otherwise it's null when format is specified in `options`.